data/transactions/transaction.go

```diff
@@ -113,6 +113,12 @@ type ApplyData struct {
 	ReceiverRewards basics.MicroAlgos `codec:"rr"`
 	CloseRewards    basics.MicroAlgos `codec:"rc"`
 	EvalDelta       EvalDelta         `codec:"dt"`
+
+	// If asa or app is being created, the id used. Else 0.
+	// Names chosen to match naming the corresponding txn.
+	// These are populated on when MaxInnerTransactions > 0 (TEAL 5)
+	ConfigAsset   basics.AssetIndex `codec:"caid"`
+	ApplicationID basics.AppIndex   `codec:"apid"`
```

Comment:
please add `omitempty`

Reply:
`omitempty` is already on the `_struct` field, so it applies to the entire struct, right?

Reply:
I think that is handled by `omitempty` on the `_struct struct{} `.  None of the rewards or closing fields use it.

Reply:
FYI, corresponding code from msgp generator
```golang
func isFieldOmitEmpty(sf StructField, s *Struct) bool {
	tagName := "omitempty"

	// go-codec distinguished between omitempty and omitemptyarray
	e := sf.FieldElem
	_, isArray := e.(*Array)
	if isArray {
		tagName = "omitemptyarray"
	}

	return sf.HasTagPart(tagName) || s.UnderscoreStructHasTagPart(tagName)
```